service/src/test/java/bio/terra/externalcreds/BaseTest.java

```diff
@@ -14,6 +15,12 @@
 @ActiveProfiles({"test", "human-readable-logging"})
 @Transactional
 @Rollback
+// We need this because Spring Boot doesn't re-use contexts when @MockBean is used in a test,
+// meaning a new connection pool is spun up per test class, causing us to blow through the max
+// allowable connections. Using the DirtiesContext annotation, we can tell Spring not to cache
+// the test context, meaning the connection pool will be shut down after each class, freeing up
+// connections. https://github.com/spring-projects/spring-boot/issues/7174
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_CLASS)
```

Comment:
Fascinating!

Reply:
Could also refactor tests so that they don't actually rely on a database :)  Eventually there should be _some_ tests that actually check that we are reading, writing, and querying correctly, but we shouldn't need to do that everywhere.